sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -530,211 +376,99 @@ class wi_element<uint16_t, NumRows, NumCols, Layout, Group> {
-        idx);
-    return *this;
-#else
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
-#endif // __SYCL_DEVICE_ONLY__
-  }
-
-  friend uint16_t
-  operator-(const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &lhs,
-            const uint16_t &rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    return make_bf16(
-        make_fp32(__spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx)) -
-        make_fp32(rhs));
-#else
-    (void)lhs;
-    (void)rhs;
-    throw runtime_error("joint matrix is not supported on host device.",
-                        PI_INVALID_DEVICE);
+#if __SYCL_DEVICE_ONLY__
+#define OP(opassign, op)                                                       \
+  wi_element &operator opassign(const uint16_t &rhs) {                         \
+    M.spvm = __spirv_VectorInsertDynamic(                                      \
+        M.spvm,                                                                \
+        make_bf16(make_fp32(__spirv_VectorExtractDynamic(M.spvm, idx)          \
+                                op make_fp32(rhs))),                           \
+        idx);                                                                  \
+    return *this;                                                              \
+  }
+#else // __SYCL_DEVICE_ONLY__
+#define OP(opassign, op)                                                       \
+  wi_element &operator opassign(const uint16_t &rhs) {                         \
+    (void)rhs;                                                                 \
+    throw runtime_error("joint matrix is not supported on host device.",       \
+                        PI_INVALID_DEVICE);                                    \
+  }
+#endif // __SYCL_DEVICE_ONLY__
+  OP(+=, +)
+  OP(-=, -)
+  OP(*=, *)
+  OP(/=, /)
+#undef OP
+
+#if __SYCL_DEVICE_ONLY__
+#define OP(type, op)                                                           \
+  friend type operator op(                                                     \
+      const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &lhs,        \
+      const uint16_t &rhs) {                                                   \
+    return make_bf16(make_fp32(                                                \
+        __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx)) op make_fp32(rhs)); \
+  }                                                                            \
+  friend type operator op(                                                     \
+      const uint16_t &lhs,                                                     \
+      const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &rhs) {      \
+    return make_bf16(make_fp32(                                                \
+        __spirv_VectorExtractDynamic(rhs.M.spvm, rhs.idx)) op make_fp32(lhs)); \
+  }
+  OP(uint16_t, +)
+  OP(uint16_t, -)
+  OP(uint16_t, *)
+  OP(uint16_t, /)
```

Comment:
What is the motivation to have the type here? Seems like it's the same for all and the other types in the definition are hard-coded anyway. Same goes for the relational operators below.